src/components/TodoItem.tsx

```diff
@@ -0,0 +1,95 @@
+import classNames from 'classnames';
+import { useContext, useEffect, useRef } from 'react';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+  editing: Todo | null,
+  onEdit: (todo: Todo | null) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  editing,
+  onEdit,
+}) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const { onDeleteTodo, onUpdateTodos } = useContext(TodosContext);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editing) {
+      inputRef.current?.focus();
+    }
+  }, [editing]);
+
+  const onEditSubmit = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+    currentId: number,
+  ) => {
+    if (event.key === 'Escape') {
+      onEdit(null);
+    }
+
+    if (event.key === 'Enter') {
```

Comment:
Would be great to use Enum instead of the hardcoded values. 
Also, switch case could be used here. You might want to add more cases later 🤔